include/deal.II/matrix_free/operators.h

```diff
@@ -205,6 +397,322 @@ namespace MatrixFreeOperators
       }
   }
 
+  //----------------- Base operator -----------------------------
+  template <int dim, typename number>
+  Base<dim,number>::~Base ()
+  {
+  }
+
+
+
+  template <int dim, typename number>
+  Base<dim,number>::Base ()
+    :
+    Subscriptor(),
+    data(NULL),
+    have_interface_matrices(false)
+  {
+  }
+
+
+
+  template <int dim, typename number>
+  typename Base<dim,number>::size_type
+  Base<dim,number>::m () const
+  {
+    Assert(data != NULL,
+           ExcNotInitialized());
+    return data->get_vector_partitioner()->size();
+  }
+
+
+
+  template <int dim, typename number>
+  typename Base<dim,number>::size_type
+  Base<dim,number>::n () const
+  {
+    return m();
+  }
+
+
+
+  template <int dim, typename number>
+  void
+  Base<dim,number>::clear ()
+  {
+    data = NULL;
```

Comment:
not sure about it, from documentation of `SmartPointer::clear()`:

> Delete the object pointed to and set the pointer to zero.

I certainly don't want to delete the object as it can be used elsewhere.